### PR TITLE
Struct inputs

### PR DESCRIPTION
**Pull request by vext01**
Companion to https://github.com/softdevteam/ykrustc/pull/132 and clean up `exec_trace` while we are here (see commit message).

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -1333,13 +1333,15 @@ mod tests {
 
     #[test]
     fn test_function_call_simple() {
-        let mut inputs = trace_inputs((0,));
+        struct IO(u8);
+
```
Comment (ptersilie) [inline ykcompile/src/lib.rs:1337]: extra newline?

File: ykcompile/src/lib.rs [hunk 2]
```diff
@@ -1620,31 +1635,33 @@ mod tests {
...
-        let mut inputs = trace_inputs((1, 2, 3));
+        struct IO(u64, u64, u64);
+        let mut inputs = trace_inputs(IO(1, 2, 3));
         let th = start_tracing(TracingKind::HardwareTracing);
         inputs.0 = unsafe { add6(inputs.0, inputs.1, inputs.2, 4, 5, 6) };
         let sir_trace = th.stop_tracing().unwrap();
         let tir_trace = TirTrace::new(&*SIR, &*sir_trace).unwrap();
-        let ct = TraceCompiler::<&(u64, u64, u64)>::compile(tir_trace);
-        let mut args = (1, 2, 3);
+        let ct = TraceCompiler::<IO>::compile(tir_trace);
```
Comment (ptersilie) [inline ykcompile/src/lib.rs:1658]: I like this. Makes it harder to make mistakes in the tests.

**Comment by ptersilie** [comment]
Looks fine.

**Comment by vext01** [comment]
Fixed.

**Comment by ptersilie** [comment]
Please squash.

**Comment by vext01** [comment]
splat.


Since the ABI doesn't change, I think we don't need to do cycle breaking?

**Comment by ptersilie** [comment]
Fair enough. Let's try!

bors r+

**Comment by ptersilie** [comment]
Hah, guess not.

**Comment by vext01** [comment]
This should be good to go.

**Pull request merged**